### PR TITLE
Support custom formatting during parsing

### PR DESCRIPTION
**Pull request by mindreframer**
Most of the code is from 
https://github.com/woahdae/simple_xlsx_reader/blob/master/lib/simple_xlsx_reader.rb#L231

I just reshuffled it a bit. 

With some unit tests, it works great for my case, but might need some checking to be generally useful. 

This fixes my biggest isseu with creek: not supporting parsing date_types and also custom formats. 

Hope, that you'll like it. 

Cheers, 
Roman

### CODE REVIEW

File: lib/creek/sheet.rb [hunk 1]
```diff
@@ -78,12 +80,13 @@ def rows_generator include_meta_data=false
                 y << (include_meta_data ? row : processed_cells)
               elsif (node.name.eql? 'c') and (node.node_type.eql? opener)
                   shared = node.attribute('t').eql? 's'
-                  cell = node.attribute('r')
+                cell_type      = node.attribute('t')
```
Comment (fatihorhan) [inline lib/creek/sheet.rb]: libxml version on my machine gives wrong values for non-existent attributes (It gives "http://schemas.openxmlformats.org/spreadsheetml/2006/main").
I had to change `node.attribute('t')` to `node.attributes['t']` so cell_type would be nil if 't' attribute doesn't exists.
This fixes my problem of dates not being parsed correctly.

**Comment by mindreframer** [comment]
@fatihorhan thx for figuring it out!

**Comment by fatihorhan** [comment]
@mindreframer thank you for sharing this.

**Comment by pythonicrubyist** [comment]
Thanks for your valuable contribution. I have merged your changes into master and published version 1.0.6.

**Comment by mindreframer** [comment]
@pythonicrubyist cool, it's merged! :dancers:

**Pull request merged**